src/docker.rs

```diff
@@ -0,0 +1,167 @@
+            Ok(id) => id,
+            Err(e) => {
+                return Err(anyhow!("failed to start the image: {}", e));
+            }
+        };
+
+        let mut content_path_buffer = PathBuf::new();
+        content_path_buffer.push(&content_path);
+
+        let mut download_path_buffer = PathBuf::new();
+        download_path_buffer.push(&download_path);
+
+        // Get the files from the container
+        let bytes = self
+            .runtime
+            .containers()
+            .get(&*container_id)
+            .copy_from(&content_path_buffer)
+            .try_concat()
+            .await?;
+
+        // Fail out if the buffer data processed is empty
+        if bytes.is_empty() {
+            return Err(anyhow!("failed to retrieve the files from the container"));
+        }
+
+        // Unpack the archive
+        let mut archive = Archive::new(&bytes[..]);
+        if write_to_stdout {
+            unimplemented!()
+        } else {
+            archive.unpack(&download_path_buffer)?;
+        }
+
+        info!(
+            "✅ Copied content to {} successfully",
+            download_path_buffer.display()
+        );
+
+        // Stop the container
+        match self.stop(container_id).await {
+            Ok(_) => {}
+            Err(e) => {
+                return Err(anyhow!("failed to stop the image: {}", e));
+            }
+        }
+
+        Ok(())
+    }
+
+    // start takes the the image struct's values to build a container
+    // by interacting the container runtime's socket.
+    async fn start(&self) -> Result<String> {
+        // note(tflannag): Use a "dummy" command "FROM SCRATCH" container images.
+        let cmd = vec![""];
+        let create_opts = ContainerCreateOpts::builder(&self.image).cmd(&cmd).build();
+        let container = self.runtime.containers().create(&create_opts).await?;
+        let id = container.id().to_string();
+
+        debug!("📦 Created container with id: {:?}", id);
+        Ok(id)
+    }
+
+    // stop takes the given container ID and interacts with the container
+    // runtime socket to stop the container.
+    async fn stop(&self, id: String) -> Result<()> {
+        let delete_opts = RmContainerOpts::builder().force(true).build();
+        if let Err(e) = self
+            .runtime
+            .containers()
+            .get(&*id)
+            .remove(&delete_opts)
+            .await
+        {
+            return Err(anyhow!("{}", e));
+        }
+
+        debug!("📦 Cleaned up container {:?} successfully", id);
+        Ok(())
+    }
+
+    async fn is_present_locally(&self) -> bool {
+        debug!("📦 Searching for image {} locally", self.image);
+        match self.runtime.images().list(&Default::default()).await {
+            Ok(images) => {
+                for image in images {
+                    if let Some(repo_tag) = image.repo_tags {
+                        for tag in repo_tag {
+                            println!("{}", tag);
```

Comment:
Do we need this `println()!` here?

Reply:
Nope, this is a debugging artifact. Good catch.